### ISSUE TITLE
MoveToParent doesn't take attribute navigation into account

### ISSUE THREAD

**Issue by martin-honnen** [AngleSharp/AngleSharp.XPath#30]
It looks as if  `MoveToParent` doesn't take any attribute navigation in the navigator into account as in that case the "move" should go back to the element parent of the attribute, i.e. in this case of having the attributes in a list and not exposed as nodes the method should check whether a move was done (which I think can be checked by comparing `_attrIndex > - 1`) and then to only reset that `_attrIndex = -1` and return true but the `_currentNode` should *not* be set to the `Parent`.

Based on that I think the method should be
```
public override bool MoveToParent()
{
            if (HasAttributes && _attrIndex > -1)
            {
                _attrIndex = -1;
                return true;
            }

	if (_currentNode.Parent == null)
	{
		return false;
	}

	_currentNode = _currentNode.Parent;
	return true;
}
```
and a test case would be

```
        [Test]
        public void TestAttributeNavigation()
        {
            var xml = @"<root att1='value 1' att2='value 2'><child>foo</child></root>";

            var parser = new XmlParser();

            var doc = parser.ParseDocument(xml);

            var nav = doc.CreateNavigator(false);
            
            Assert.AreEqual(nav.Name, "root");
            if (nav.MoveToFirstAttribute())
            {
                do
                {
                    Assert.AreEqual(nav.NodeType, XPathNodeType.Attribute);
                }
                while (nav.MoveToNextAttribute());
                nav.MoveToParent();
            }
            Assert.AreEqual(nav.Name, "root");
            
        }
```

**Comment by martin-honnen** [comment]
The 2 alpha release still throws off any attribute based navigation due to lack of attribute treatment in `MoveToParent`; I think it should now do something like

```
public override bool MoveToParent()
{
        if (_currentNode is IAttr attr)
        {
           _currentNode = attr.OwnerElement;
           return true;
        }

	if (_currentNode.Parent == null)
	{
		return false;
	}

	_currentNode = _currentNode.Parent;
	return true;
}
```

**Comment by martin-honnen** [comment]
My main problem has been using `ReadSubtree` to be able to feed an AngleSharp HTML or XML DOM document to older APIs which only take an `XmlReader`; with the change suggested to `MoveToParent()` above I think that works fine now, a rather complex test case would be

```
       public void TestReadSubTree()
        {
            var xml = @"<!DOCTYPE html><html lang='en'><head><meta charset='utf-8'/><title>Test</title></head><body><h1>Test</h1><p id='p-1'>This is paragraph 1 line 1.<br/>Line 2.</p><p>This is paragraph 2 line 1.<br/>Line 2.</p></body></html>";

            var xmlDoc = new XmlParser().ParseDocument(xml);

            var angleSharpNav = xmlDoc.CreateNavigator(false);

            var xDoc = XDocument.Parse(xml);

            var xNav = xDoc.Root.CreateNavigator();

            using (var angleSharpReader = angleSharpNav.ReadSubtree())
            {
                using (var xReader = xNav.ReadSubtree())
                {
                    while (angleSharpReader.Read() && xReader.Read())
                    {
                        Assert.AreEqual(angleSharpReader.NodeType, xReader.NodeType);
                        Assert.AreEqual(angleSharpReader.LocalName, xReader.LocalName);
                        Assert.AreEqual(angleSharpReader.NamespaceURI, xReader.NamespaceURI);
                        Assert.AreEqual(angleSharpReader.HasAttributes, xReader.HasAttributes);

                        if (angleSharpReader.NodeType == XmlNodeType.Element && angleSharpReader.HasAttributes && xReader.HasAttributes)
                        {
                            while (angleSharpReader.MoveToNextAttribute() && xReader.MoveToNextAttribute())
                            {
                                Assert.AreEqual(angleSharpReader.NodeType, xReader.NodeType);
                                Assert.AreEqual(angleSharpReader.LocalName, xReader.LocalName);
                                Assert.AreEqual(angleSharpReader.NamespaceURI, xReader.NamespaceURI);
                            }

                            angleSharpReader.MoveToElement();
                            xReader.MoveToElement();
                        }
                    }
                }
            }
        }

```

**Comment by denis-ivanov** [comment]
Fixed `2.0.0-alpha-3`.
